src/main/java/ch/njol/skript/SkriptEventHandler.java

```diff
@@ -212,62 +215,86 @@ static ScriptInfo removeTriggers(File script) {
 				i--;
 			}
 		}
-		
+
 		info.commands = Commands.unregisterCommands(script);
-		
+
 		return info;
 	}
-	
+
 	static void removeAllTriggers() {
 		triggers.clear();
 		for (Trigger t : selfRegisteredTriggers)
 			((SelfRegisteringSkriptEvent) t.getEvent()).unregisterAll();
 		selfRegisteredTriggers.clear();
 	}
-	
+
 	/**
 	 * Registers event handlers for all events which currently loaded
 	 * triggers are using.
 	 */
-	@SuppressWarnings({"unchecked", "rawtypes"})
 	static void registerBukkitEvents() {
 		for (NonNullPair<Class<? extends Event>, Trigger> pair : triggers) {
 			assert pair.getFirst() != null;
 			Class<? extends Event> e = pair.getFirst();
-			
+
 			EventPriority priority;
 			priority = pair.getSecond().getEvent().getEventPriority();
-			
+
 			PriorityListener listener = listeners[priority.ordinal()];
 			EventExecutor executor = listener.executor;
-			
-			Set<Class<? extends Event>> registeredEvents = listener.registeredEvents;
-			
+
+			HandlerList handlerList = getHandlerList(e);
+
+			if (handlerList == null)
+				continue;
+
 			// PlayerInteractEntityEvent has a subclass we need for armor stands
 			if (e.equals(PlayerInteractEntityEvent.class)) {
-				if (!registeredEvents.contains(e)) {
-					registeredEvents.add(e);
+				if (!isEventRegistered(handlerList, priority)) {
 					Bukkit.getPluginManager().registerEvent(e, listener, priority, executor, Skript.getInstance());
 					Bukkit.getPluginManager().registerEvent(PlayerInteractAtEntityEvent.class, listener, priority, executor, Skript.getInstance());
 				}
 				continue;
 			}
-			if (e.equals(PlayerInteractAtEntityEvent.class) || e.equals(PlayerArmorStandManipulateEvent.class)) {
+			if (e.equals(PlayerInteractAtEntityEvent.class) || e.equals(PlayerArmorStandManipulateEvent.class))
 				continue; // Ignore, registered above
-			}
-			
-			if (!containsSuperclass((Set) registeredEvents, e)) { // I just love Java's generics
+
+			if (!isEventRegistered(handlerList, priority)) // Check if event is registered
 				Bukkit.getPluginManager().registerEvent(e, listener, priority, executor, Skript.getInstance());
-				registeredEvents.add(e);
+		}
+	}
+
+	@SuppressWarnings("ThrowableNotThrown")
+	@Nullable
```

Comment:
`@Override -> @Nullable > @SuppressWarnings`
```suggestion
	@Nullable
	@SuppressWarnings("ThrowableNotThrown")
```